Trackers/centroid_tracker/centroid.py

```diff
@@ -1,30 +1,112 @@
+        - else, find the min. distance of old centroid n new centroid
+            - check whether the centroid is matched to any row or col, if not then this means this object was lost earlier and now found, and reset it miss to 0
+            - check if input centroids is less than last time, so some objects are lost, and icrease their miss and check if it >max_age deregister it.
+                - if not, then we new objects, so register them
+        '''
+        if(detections==[] and self.tracks==[]):
+            return self.tracks
+        if(detections==[] and self.tracks!=[]):
+            print(id)
+            # increase every objects miss by 1
+            for track in self.tracks:
+                track.miss +=1
+            self.delete_track()
+            return self.tracks
+
+        inputCentroids = np.zeros((max(1, len(detections)), 2), dtype="int")
 
-    
\ No newline at end of file
+        for i in range(0,len(detections)):
+            cX = int((detections[i][2][0]+ detections[i][2][1]) / 2.0)
+            cY = int((detections[i][2][2] + detections[i][2][3]) / 2.0)
+            inputCentroids[i] = (cX, cY)
+        
+        objectId = []
+        objectCentroid = []
+        if len(self.tracks) == 0:
+            for i in range(0, len(inputCentroids)):
+                self.add_track(self.nextID, detections,inputCentroids[i])
+        else:      
+            for track in self.tracks:
+                objectId.append(track.id)
+                objectCentroid.append(track.centroid) 
+            D = dist.cdist(np.array(objectCentroid), inputCentroids)
+            rows = D.min(axis=1).argsort()
+            cols = D.argmin(axis=1)[rows]
+            usedRows = set()
+            usedCols = set()
+            # loop over the combination of the (row, column) index tuples
+            for (row, col) in zip(rows, cols):
+                # if we have already examined either the row or column value before, ignore it val
+                if row in usedRows or col in usedCols:
+                    continue
+                self.tracks[row].centroid = inputCentroids[col]
+                self.tracks[row].miss = 0
+                # indicate that we have examined each of the row and column indexes, respectively
+                usedRows.add(row)
+                usedCols.add(col)
+            unusedRows = set(range(0, D.shape[0])).difference(usedRows)
+            unusedCols = set(range(0, D.shape[1])).difference(usedCols)
+            if D.shape[0] >= D.shape[1]:
+            # loop over the unused row indexes
+                for row in unusedRows: 
+                    # grab the object ID for the corresponding row index and increment the disappeared counter
+                    self.tracks[row].miss += 1
+                    # check to see if the number of consecutive frames the object has been marked "disappeared" for warrants deregistering the object
+                    if self.tracks[row].miss >= self.max_age:
```

Comment:
`self.delete()` function already takes into account this condition.

Reply:
we can do this, but with this it will only call `delete_track()` when `self.miss >= max_age` instead of every time calling `delete_track()`

Reply:
You don't required to call `delete_track()` every time, you just have to call just before returning from `update()` method.

Reply:
yes, understood